packages/zoe/src/contracts/vpool-xyk-amm/multipoolMarketMaker.js

```diff
@@ -0,0 +1,193 @@
+ * publicFacet operations support querying prices and the sizes of pools. New
+ * Pools can be created with addPool().
+ *
+ * When making trades or requesting prices, the caller must specify either a
+ * maximum input amount (swapIn, getInputPrice) or a minimum output amount
+ * (swapOut, getOutPutPrice) or both. For swaps, the required keywords are `In`
+ * for the trader's `give` amount, and `Out` for the trader's `want` amount.
+ * getInputPrice and getOutputPrice each take two Amounts. The price functions
+ * return both amountIn (which may be lower than the original amount) and
+ * amountOut (which may be higher). When both prices are specified, no swap will
+ * be made (and no price provided) if both restrictions can't be honored.
+ *
+ * When adding and removing liquidity, the keywords are Central, Secondary, and
+ * Liquidity. adding liquidity has Central and Secondary in the `give` section,
+ * while removing liquidity has `want` and `give` swapped.
+ *
+ * Transactions that don't require an invitation include addPool and the
+ * queries: getInputPrice, getOutputPrice, getPoolAllocation,
+ * getLiquidityIssuer, and getLiquiditySupply.
+ *
+ * @type {ContractStartFn}
+ */
+const start = zcf => {
+  // This contract must have a "Central" keyword and issuer in the
+  // IssuerKeywordRecord.
+  // poolFee is the portion of the fees that go into the pool (in BP).
+  // protocolFee is the portion of the fees that are shared with validators
+  const {
+    brands: { Central: centralBrand },
+    timer,
+    poolFee,
+    protocolFee,
+  } = zcf.getTerms();
+  assertIssuerKeywords(zcf, ['Central']);
+  assert(centralBrand !== undefined, X`centralBrand must be present`);
+
+  /** @type {WeakStore<Brand,XYKPool>} */
+  const secondaryBrandToPool = makeWeakStore('secondaryBrand');
+  const getPool = secondaryBrandToPool.get;
+  const initPool = secondaryBrandToPool.init;
+  const isSecondary = secondaryBrandToPool.has;
+
+  const quoteIssuerKit = makeIssuerKit('Quote', AssetKind.SET);
+
+  // For now, this seat collects protocol fees. It needs to be connected to
+  // something that will extract the fees.
+  const { zcfSeat: protocolSeat } = zcf.makeEmptySeatKit();
+
+  const getLiquiditySupply = brand => getPool(brand).getLiquiditySupply();
+  const getLiquidityIssuer = brand => getPool(brand).getLiquidityIssuer();
+  const addPool = makeAddPool(
+    zcf,
+    isSecondary,
+    initPool,
+    centralBrand,
+    timer,
+    quoteIssuerKit,
+    protocolFee,
+    poolFee,
+    protocolSeat,
+  );
+  const getPoolAllocation = brand => {
+    return getPool(brand)
+      .getPoolSeat()
+      .getCurrentAllocation();
+  };
+
+  const getPriceAuthorities = brand => {
+    const pool = getPool(brand);
+    return {
+      toCentral: pool.getToCentralPriceAuthority(),
+      fromCentral: pool.getFromCentralPriceAuthority(),
+    };
+  };
+
+  /**
+   * @callback ProvideVPool
```

Comment:
This stray `@callback` disconnects all the declarations in this comment from the function below.

```suggestion
```